templates/patterns/button/button.ui_patterns.yml

```diff
@@ -23,30 +23,35 @@ button:
     splash_page:
       label: "Splash page"
       description: "Special buttons used on the site's splash page."
+    ghost:
+      label: "Ghost"
+      description: "Special buttons used on the site's splash page."
   fields:
+    label:
+      type: "text"
+      label: "Label"
+      description: "Label of the button."
+      preview: "Button"
     type_attribute:
       type: "text"
       label: "Type attribute"
       description: "The buttons type attribute"
       preview: "submit"
-    href:
-      type: "Url"
-      label: "URL"
-      description: "URL to use if button needs to be a link (A Drupal Url object)"
-      preview: "http://www.example.com"
     icon:
-      type: "text"
+      type: "array"
       label: "Icon"
-      description: "An icon to use with the button."
-      preview: "caret-right"
-    label:
+      description: "An icon."
+      preview:
+        - path: ''
+        - type: 'ui'
+        - name: 'corner-arrow'
```

Comment:
In the new ECL 2 icons are, indeed, hashes but for us it's a string, we will need to keep this a string in the pattern definition and turn it into an array "behind the scenes", in the Twig template, otherwise we will not be able to upgrade seamlessly anymore.

The UI Patterns are our API, we can only add things, not change it (unless really really necessary, which is not this case).